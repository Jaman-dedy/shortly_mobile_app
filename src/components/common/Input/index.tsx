/* eslint-disable react-native/no-inline-styles */
import React from 'react';
import {useTheme} from '@shopify/restyle';
import {TextInput, StyleSheet} from 'react-native';

import {Theme} from '@app/assets/style/theme';

type inputProps = {
  placeholder?: string;
  formData: Record<string, string>;
  errors: Record<string, string>;
  onChangeText: (args: {name: string; value: string}) => void;
};

const Input = ({placeholder, formData, onChangeText, errors}: inputProps) => {
  const theme = useTheme<Theme>();
  const styles = StyleSheet.create({
    input: {
      ...theme.inputVariant,
      borderColor: errors?.message ? 'red' : 'white',
      color: errors?.message ? 'red' : 'black',
      backgroundColor: 'white',
      textAlign: 'center',
    },
  });
  return (
    <TextInput
      value={errors?.message ? errors.message : formData?.link}
      onChangeText={(linkValue: string): void =>
        onChangeText({name: 'link', value: linkValue})
      }
      style={styles.input}
      placeholder={placeholder}
      testID="text-input"
    />
  );
};

export default Input;
