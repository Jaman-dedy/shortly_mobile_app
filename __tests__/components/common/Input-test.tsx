import React from 'react';
import renderer from 'react-test-renderer';
import {cleanup} from '@testing-library/react-native';

import Input from '../../../src/components/common/Input';

import MockThemes from '../../../mock/MockTheme';

afterEach(cleanup);

describe('Text Input component', () => {
  const mockPlaceholder = 'shorten a link here';
  const mockData = {
    link: 'www.example.com',
  };
  const mockError = {
    error: 'invalid url',
  };
  const mockInput = jest.fn();
  test('renders correctly', async () => {
    const tree = await renderer
      .create(
        <MockThemes>
          <Input
            placeholder={mockPlaceholder}
            formData={mockData}
            errors={mockError}
            onChangeText={mockInput}
          />
        </MockThemes>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
