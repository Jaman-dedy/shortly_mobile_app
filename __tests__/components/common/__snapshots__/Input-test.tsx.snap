// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Input component renders correctly 1`] = `
<TextInput
  onChangeText={[Function]}
  placeholder="shorten a link here"
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "white",
      "borderRadius": 4,
      "borderWidth": 1,
      "color": "black",
      "fontSize": 18,
      "height": 45,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
      "textAlign": "center",
      "width": 280,
    }
  }
  testID="text-input"
  value="www.example.com"
/>
`;
